src/timely-util/src/activator.rs

```diff
@@ -224,3 +225,30 @@ impl ArcActivatorInner {
         self.activated = 0;
     }
 }
+
+/// An activator that ignores new activations if one is already pending.
+pub struct LimitingActivator {
+    inner: Activator,
+    next_activation: Instant,
+}
+
+impl LimitingActivator {
+    pub fn new(inner: Activator) -> Self {
+        Self {
+            inner,
+            next_activation: Instant::now(),
+        }
+    }
+
+    pub fn activate(&mut self) {
+        self.activate_after(Duration::ZERO);
+    }
+
+    pub fn activate_after(&mut self, duration: Duration) {
+        let now = Instant::now();
+        if self.next_activation < now {
```

Comment:
Nit: this should probably be <= although it's very unlikely to precisely hit a specific point in time?

Reply:
Probably! I guess it is possible for the activator to trigger the pending activation at exactly `next_activation` and for `activate_after` to be called at exactly the same time. I'll change that.